main.rb

```diff
@@ -1,44 +1,250 @@
 
 module Enumerable
-# 1.my_each
+  # 1.my_each
   def my_each
-    #return to_enum(:my_each) unless block_given?
+    # return to_enum(:my_each) unless block_given?
     if block_given?
       a = 0
-      while a < self.length
-        yield self[a]
+      while a < to_a.length
+        yield to_a[a]
         a += 1
       end
-      self 
-    else 
+      self
+    else
       to_enum(:my_each)
-    end  
+    end
   end
-  
-# 2.my_each_index 
+
+  # 2.my_each_index
   def my_each_with_index
-    #return to_enum(:my_each_index) unless block_given?
+    # return to_enum(:my_each_index) unless block_given?
     if block_given?
       a = 0
-      while a < self.length
-        yield(self[a], a)
+      while a < to_a.length
+        yield(to_a[a], a)
         a += 1
       end
-      self 
-    else 
+      self
+    else
       to_enum(:my_each_with_index)
-    end  
+    end
   end
 
-# 2.my_select
+  # 2.my_select
   def my_select
-    #return to_enum(:my_each_index) unless block_given?
+    # return to_enum(:my_each_index) unless block_given?
+    if block_given?
+      new_arr = []
+      to_a.my_each { |item| new_arr << item if yield(item) }
+      new_arr
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  # 4.my_all?
+  def my_all?(argument = nil)
+    index = 0
+    while index < length
+      if block_given?
+        return false if yield(self[index]) == false
+      else
+        if argument.nil? # when there is no argument
+          return false if self[index].nil? || self[index] == false
+        elsif argument.class == Regexp # when there is an argument
+          return false if self[index].match?(argument) != true
+        elsif argument.class == Class
+          return false if self[index].is_a?(argument) != true
+        else # when the argument is no regexp ni class
+          return false if self[index] != argument
+        end
+      end
+      index += 1
+    end
+    true
+  end
+
+  # 4.my_any?
+  def my_any?(arg = nil)
+    a = 0
+    o = false
+    my = to_a
+    if block_given?
+      while a < my.length
+        if yield(my[a]) == true
+          o = true
+          break
+        end
+        a += 1
+      end
+      o
+    elsif arg.nil?
+      o = true
+    elsif block_given? == true && !arg.nil?
+      o = false
+      while a < my.length
+        if (my[a].is_a? arg) == true
+          o = true
+          break
+        end
+        a += 1
+      end
+      o
+    elsif arg.class == Regexp
+      while a < my.length
+        if !my[a].match(arg).nil?
+          o = true
+          break
+        else
+          o = false
+        end
+        a += 1
+      end
+      o
+    elsif arg.class != Regexp && arg.class == String
+      while a < my.length
+        if my[a] == arg
+          o = true
+          break
+        else
+          o = false
+        end
+        a += 1
+      end
+      o
+    else
+      true
+    end
+  end
```

Comment:
- [ ] When no block or argument is given, `my_any?` should return false if at least one of the members of the collection is not true. 

```ruby
array = [nil, false]
array.my_any? === array.any? => false
```

- [ ] When a class is passed as an argument, `my_any?` should return true if at least one of the members of the collection is a member of that class. 
```ruby
words = %w[dog door rod blade]
words.my_any?(Integer) === words.any?(Integer) => false
```